### PR TITLE
Deprecate date aggregations on boolean fields

### PR DESCRIPTION
**Pull request by not-napoleon**
Related to #59255

This PR would begin to issue a deprecation warning when running a date range, date histogram, or auto date histogram over a boolean field.  This is currently permitted, but is unlikely to be what the user intended to do, as a histogram over the values [0, 1] is not terribly useful.  One possible way users could encounter this situation would be an index pattern that includes indexes with different mappings.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/DateRangeAggregationBuilder.java [hunk 1]
```diff
@@ -45,8 +50,49 @@
         }, (p, c) -> RangeAggregator.Range.PARSER.parse(p, null), RangeAggregator.RANGES_FIELD);
     }
 
+    private static final DeprecationLogger DEPRECATION_LOGGER = DeprecationLogger.getLogger(AbstractRangeAggregatorFactory.class);
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/DateRangeAggregationBuilder.java:53]: Should this be `DateRangeAggregatorBuilder`?

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregatorFactory.java [hunk 2]
```diff
@@ -25,15 +27,57 @@
 import java.util.Map;
 
 public final class DateHistogramAggregatorFactory extends ValuesSourceAggregatorFactory {
+    private static final DeprecationLogger DEPRECATION_LOGGER = DeprecationLogger.getLogger(DateHistogramAggregator.class);
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregatorFactory.java:30]: Should this be the factory? I donno.

**Comment by not-napoleon** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by not-napoleon** [comment]
@elasticmachine update branch

**Review by nik9000 — APPROVED** [review]
LGTM, though I think it's worth adding a unit test for it.

**Comment by not-napoleon** [comment]
@elasticmachine update branch

**Pull request merged**